### PR TITLE
pool: Optimizing amount of updateInterestRate calls

### PR DESCRIPTION
**Pull request by mijovic**

### CODE REVIEW

File: contracts/amm/TempusAMM.sol [hunk 1]
```diff
@@ -651,7 +651,8 @@ contract TempusAMM is BaseMinimalSwapInfoPool, StableMath, IRateProvider {
     function _getTokenRates() private returns (uint256[] memory) {
         uint256[] memory rates = new uint256[](_TOTAL_TOKENS);
         rates[0] = _token0.getPricePerFullShare();
-        rates[1] = _token1.getPricePerFullShare();
+        // We already did updateInterestRate, so we can use stored values
```
Comment (RedFox20) [inline contracts/amm/TempusAMM.sol:654]: Nice! 🧠

File: contracts/TempusPool.sol [hunk 2]
```diff
@@ -422,15 +418,17 @@ abstract contract TempusPool is ITempusPool, ReentrancyGuard, Ownable, Versioned
     }
 
     function pricePerYieldShare() external override returns (uint256) {
-        return pricePerYieldShare(currentYield(), estimatedYield());
+        uint256 yield = currentYield();
+        return pricePerYieldShare(yield, estimatedYield(yield));
     }
 
     function pricePerYieldShareStored() external view override returns (uint256) {
         return pricePerYieldShare(currentYieldStored(), estimatedYieldStored());
```
Comment (RedFox20) [inline contracts/TempusPool.sol:426]: Maybe update this as well so the code looks similar to your updates? Less calls should reduce gas a bit as well.
```ts
uint256 yield = currentYieldStored();
return pricePerYieldShare(yield, estimatedYield(yield));
```

File: contracts/amm/TempusAMM.sol (hunk 1 above)
Comment (axic) [reply contracts/amm/TempusAMM.sol:654]: From a clarity perspective, having `pool.updateInterest()` and then using `_token0.getPricePerFullShare(); _token1.getPricePerFullShare();` would be cool.

Comment (RedFox20) [reply contracts/amm/TempusAMM.sol:654]: Expanding on your idea, the most optimal way to get those rates would be to have:
`tempusPool.getTokenPricesPerFullShare()` and `tempusPool.getTokenPricesPerFullShareStored()`

File: contracts/TempusPool.sol (hunk 2 above)
Comment (mijovic) [reply contracts/TempusPool.sol:426]: We can do this in separate PR

File: contracts/amm/TempusAMM.sol (hunk 1 above)
Comment (mijovic) [reply contracts/amm/TempusAMM.sol:654]: This can be done in separate PR

**Pull request merged**